extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api-sagemaker/src/main/resources/docker/Dockerfile

```diff
@@ -1,37 +1,43 @@
-
 ARG DOCKER_BASELINE_REPO_ID
 ARG VERSION_AISSEMBLE
-FROM ${DOCKER_BASELINE_REPO_ID}boozallen/aissemble-fastapi:${VERSION_AISSEMBLE} AS builder
 
-RUN python -m venv /opt/venv
-ENV PATH="/opt/venv/bin:$PATH"
+#HABUSHU_BUILDER_STAGE - HABUSHU GENERATED CODE (DO NOT MODIFY)
+FROM python:3.11-slim AS habushu_builder
+# Poetry and supporting plugin installations
+RUN python -m ensurepip --upgrade && \
+    pip install poetry && \
+    poetry self add poetry-monorepo-dependency-plugin && \
+    poetry self add poetry-plugin-bundle
 
-COPY ./target/wheels/* /tmp/wheels/
-# Re-install any dependencies defined in the base image into the virtual environment, then install new requirements
-RUN pip install -r /tmp/requirements.txt && \
-    set -e && \
-    cd /tmp/wheels/; for x in *.whl; do pip install $x --no-cache-dir; done
+WORKDIR /work-dir
+COPY --chown=1001 target/containerize-support ./containerize-support/
+RUN find . -type f -name pyproject.toml -exec sed -i 's|develop[[:blank:]]*=[[:blank:]]*true|develop = false|g' {} \;
 
+USER root
+WORKDIR /work-dir/containerize-support/extensions/aissemble-extensions-model-training-api-sagemaker
+ENV POETRY_CACHE_DIR="/.cache/pypoetry"
 
-# Start a new build stage based on a slim Python image, which will have the minimal amount of code, FastAPI
-# components, and installed dependencies copied into it from the initial builder image
-FROM python:3.11-slim as final
+# export target project's virtual environment to /opt/venv
+RUN --mount=type=cache,target=/.cache/pypoetry/ \
+    poetry lock && \
+    poetry bundle venv /opt/venv
+#HABUSHU_BUILDER_STAGE - HABUSHU GENERATED CODE (END)
 
-LABEL org.opencontainers.image.source = "https://github.com/boozallen/aissemble"
-
-WORKDIR /app
+FROM ${DOCKER_BASELINE_REPO_ID}boozallen/aissemble-fastapi:${VERSION_AISSEMBLE} AS builder
 
-# Set/move any required environmental variables and key scripts/configs needed by FastAPI from the base image
-# into the slimmed down final Python image
-ENV API_MAIN model_training_api_sagemaker
+#HABUSHU_FINAL_STAGE - HABUSHU GENERATED CODE (DO NOT MODIFY)
+FROM python:3.11-slim AS final
+# Copy venv from builder and activate by adding to the path
+COPY --from=habushu_builder --chown=1001 /opt/venv /opt/venv/
 ENV PATH="/opt/venv/bin:$PATH"
-ENV PYTHONPATH=/app
-COPY --from=builder /start.sh /start.sh
+#HABUSHU_FINAL_STAGE - HABUSHU GENERATED CODE (END)
 
-# Copy over the virtual environment (and all pip installed packages) from the initial builder stage
-COPY --from=builder /opt/venv /opt/venv
+LABEL org.opencontainers.image.source="https://github.com/boozallen/aissemble"
 
-COPY ./target/model_training_api_sagemaker.py /app/model_training_api_sagemaker.py
+# Set/move any required environmental variables and key scripts/configs needed by FastAPI from the base image
+# into the slimmed down final Python image
+ENV API_MAIN=model_training_api_sagemaker.model_training_api_sagemaker
+COPY --from=builder --chown=1001 /start.sh /start.sh
```

Comment:
I: As I was updating the image to work with the new Habushu feature (including running the APIs as a module instead of directly via a `.py` file) I noticed that we're largely just using the `aissemble-fastapi` image as a way to pass this start script.  At the end of the day, these images have the same base as `aissemble-fastapi`, so I think we could probably just use that image for the `dockerBase` configuration of Habushu?  However, it also sets up an example API server which complicated things a bit.  This is at least better than we found it.